peps/pep-0747.rst

```diff
@@ -711,12 +735,38 @@ assigned to variables and manipulated like any other data in a program:
 ``TypeExpr[]`` is how you spell the type of a variable containing a
 type annotation object describing a type.
 
-``TypeExpr[]`` is similar to ``type[]``, but ``type[]`` can only used to
+``TypeExpr[]`` is similar to ``type[]``, but ``type[]`` can only
 spell simple **class objects** like ``int``, ``str``, ``list``, or ``MyClass``.
 ``TypeExpr[]`` by contrast can additionally spell more complex types, 
 including those with brackets (like ``list[int]``) or pipes (like ``int | None``),
 and including special types like ``Any``, ``LiteralString``, or ``Never``.
 
+A ``TypeExpr`` variable looks similar to a ``TypeAlias`` definition, but
+can only be used where a dynamic value is expected.
+``TypeAlias`` (and the ``type`` statement) by contrast define a name that can
+be used where a fixed type is expected:
+
+-  Okay, but discouraged in Python 3.12+:
+
+   ::
+
+      MaybeFloat: TypeAlias = float | None
+      def sqrt(n: float) -> MaybeFloat: ...
+
+-  Yes:
+
+   ::
+
+      type MaybeFloat = float | None
+      def sqrt(n: float) -> MaybeFloat: ...
+
+-  No:
+
+   ::
+
+      maybe_float: TypeExpr = float | None
```

Comment:
Ah, I think I now understand what you were trying to say above. This is all a (confusing) way to reiterate that a variable cannot be used in a type expression. That rule is already spelled out clearly in the "Type Annotations" section of the spec, so I don't think it needs to be repeated in this PEP.

Reply:
2 non-expert commentators have been confused about when to use `TypeAlias` vs when to use `TypeExpr`, since either can be used in certain contexts. This subsection hopes to clarify.

Reply:
I understand the intent, but I don't think it's achieving the clarity you're seeking. I'm finding it to be very confusing. I think it could be made clear if it were reworked, but I'm not convinced this is needed.

Type alias definitions and the `TypeExpr` symbol are *very* different, so there should be no confusion between the two. The current name (`TypeExpr`) is likely contributing to the confusion. Perhaps once the name is fixed, it will become clearer to non-expert readers.